--- a/include/h5rd/detail/Node_detail.h
+++ b/include/h5rd/detail/Node_detail.h
@@ -114,7 +114,7 @@
     for (std::size_t i = 0; i < group_info.nlinks; ++i) {
         H5O_info_t oinfo{};
         herr_t errorCode;
-        # if H5_VERS_MAJOR >= 1 && H5_VERS_MINOR >= 12
+        # if (H5_VERS_MAJOR > 1) || (H5_VERS_MAJOR == 1 && H5_VERS_MINOR >= 12)
         errorCode = H5Oget_info_by_idx(id, ".", H5_INDEX_NAME, H5_ITER_INC, i, &oinfo, H5O_INFO_BASIC, H5P_DEFAULT);
         # else
         errorCode = H5Oget_info_by_idx(id, ".", H5_INDEX_NAME, H5_ITER_INC, i, &oinfo, H5P_DEFAULT);
